### PR TITLE
Implement `delete_traces`

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/11602?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11602/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11602
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Implement `delete_traces`.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

```python
import mlflow
from mlflow.entities.trace_data import TraceData

mlflow.set_tracking_uri("databricks")
name = "/Users/harutaka.kawamura@databricks.com/test2"
mlflow.set_experiment(name)
exp = mlflow.get_experiment_by_name(name)

client = mlflow.MlflowClient()
trace_info = client.create_trace(
    experiment_id=exp.experiment_id,
    timestamp_ms=5,
    execution_time_ms=5,
    status="OK",
)
print(trace_info)
client._tracking_client._upload_trace_data(
    trace_info.request_id,
    trace_data=TraceData(spans=[]),
)

print("deleting")
client.delete_traces(
    experiment_id=exp.experiment_id,
    request_ids=[trace_info.request_id],
)

try:
    client.get_trace_info(trace_info.request_id)
except Exception as e:
    print(e)
else:
    assert False, "Expected exception"


trace_info = client.create_trace(
    experiment_id=exp.experiment_id,
    timestamp_ms=5,
    execution_time_ms=5,
    status="OK",
)
print(trace_info)
client._tracking_client._upload_trace_data(
    trace_info.request_id,
    trace_data=TraceData(spans=[]),
)

print("deleting")
client.delete_traces(
    experiment_id=exp.experiment_id,
    max_timestamp_millis=3,
)

client.get_trace_info(trace_info.request_id)

```

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
Adding unit tests now.

File: mlflow/tracking/client.py [hunk 1]
```diff
@@ -426,6 +426,30 @@ def create_trace(
...
+    ) -> None:
+        """
+        Delete traces based on the specified criteria.
+
+        Args:
+            experiment_id: ID of the associated experiment.
+            max_timestamp_millis: The maximum timestamp in milliseconds since the UNIX epoch for
+                deleting traces.
+            max_traces: The maximum number of traces to delete.
+            request_ids: A set of request IDs to delete.
```
Comment (dbczumar) [inline mlflow/tracking/client.py:444]: Can we indicate that either `max_timestamp_millis` or `request_ids` must be specified, but not both? Can we also indicate that `max_traces` can't be specified if `request_ids` is specified?

**Review by dbczumar — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/11602/files#r1548956042 is addressed. Thanks @harupy ! :D

File: mlflow/tracking/client.py (hunk 1 above)
Comment (harupy) [reply mlflow/tracking/client.py:444]: /ai Update the docstring based on the feedback.

Comment (harupy) [reply mlflow/tracking/client.py:444]: /ai Let's mention these requirements in the function description, not parameter description.

**Pull request merged**